programs/test/benchmark.c

```diff
@@ -186,7 +186,12 @@ do {                                                                    \
         CODE;                                                           \
     }                                                                   \
                                                                         \
-    if( ret != 0 )                                                      \
+    if( ret == MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED )               \
```

Comment:
Makes sense. Does this mean that `MBEDTLS_PLATFORM_C` is non-optional everywhere after #2124 is merged? (#2124 contains the change that makes it unconditional here).

Reply:
Hmm, `MBEDTLS_PLATFORM_C` should remain optional. What's the problem that makes it unconditional? Cc @RonEld

Reply:
I can make the same change as in #2124 if so desired, the conflict would be trivial to resolve, if not automatic. I was just asking whether every source file that could receive a `MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED` will have to #include this unconditionally (which would make it basically mandatory pretty much everywhere). My own builds don't break and I can't see the CI logs: does it break because it's currently conditional, or does it break after you made the change to unconditional?

Reply:
To reproduce the failure, check out the latest `development` somewhere and run `tests/scripts/all.sh test_no_platform`. (The test is already present in your branch, but we've made a recent change to `all.sh` that makes it easy to run individual tests.)

You need to `#include <mbedtls/platform.h>` unconditionally if your code uses `MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED`.

Reply:
After making that unconditional (b81034a83b507eab6ed85d8b4b535936ae5b4701), that test complains about `mbedtls_snprintf` being redefined on my system, but #2124 does not remove it, so I'll leave it as is. Without that `#define` it builds and passes all tests on my system.